assets/js/atomic/utils/register-block-single-product-template.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { getBlockType } from '@wordpress/blocks';
+import { subscribe, select } from '@wordpress/data';
+
+export const registerBlockSingleProductTemplate = ( {
+	registerBlockFn,
+	unregisterBlockFn,
+	blockName,
+}: {
+	registerBlockFn: () => void;
+	unregisterBlockFn: () => void;
+	blockName: string;
+} ) => {
+	let currentTemplateId: string | undefined;
+
+	subscribe( () => {
```

Comment:
Could we use debounce here to make this trigger less frequent?

Reply:
Mmm, I would avoid a debounce here to avoid any kind of race condition.

Reply:
I'm curious what can be the race condition and its drawback. The motivation behind my suggestion is to keep the editor performant.

Reply:
Mmm, for example, the user is editing another template, but the function isn't called due to the debounce and the block is still visible in the inserter.

Reply:
> but the function isn't called due to the debounce

I don't think we will have this issue, as we just want to debounce them (by 100ms, for example), the subscription still triggers, but less frequently. So when the user switches to another template, the subscription may not trigger right away, but still fast enough to register/unregister the block before user interaction. We don't have to do it now btw, we can go with the current PR and come back to this later when we see it affect the editor performance.